src/security-council-mgmt/SecurityCouncilManager.sol

```diff
@@ -0,0 +1,266 @@
+                membersToRemove[0] = _member;
+                _dispatchUpdateMembers(membersToAdd, membersToRemove);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /// @notice Security council member can rotate out their address for a new one.
+    /// Rotation must be initiated by the security council, and member rotating out must give explicit
+    /// consent via signature
+    /// @param _currentAddress Address to rotate out
+    /// @param _newAddress Address to rotate in
+    /// @param _signature Signature from _currentAddress
+    function rotateMember(address _currentAddress, address _newAddress, bytes memory _signature)
+        external
+        onlyRole(MEMBER_ROTATOR_ROLE)
+    {
+        require(
+            !SecurityCouncilMgmtUtils.isInArray(_newAddress, marchCohort)
+                && !SecurityCouncilMgmtUtils.isInArray(_newAddress, septemberCohort),
+            "SecurityCouncilManager: new member already included"
+        );
+        address[] storage cohort;
+        if (SecurityCouncilMgmtUtils.isInArray(_currentAddress, marchCohort)) {
+            cohort = marchCohort;
+        } else if (SecurityCouncilMgmtUtils.isInArray(_currentAddress, septemberCohort)) {
+            cohort = septemberCohort;
+        } else {
+            revert("SecurityCouncilManager: current address not in either cohort");
+        }
+
+        // TODO: double check that this makes sense
+        bytes32 data = getRotateDataToSign(_currentAddress, _newAddress);
+        require(
+            _verify(data, _signature, _currentAddress), "SecurityCouncilManager: invalid signature"
+        );
+        for (uint256 i = 0; i < cohort.length; i++) {
+            if (cohort[i] == _currentAddress) {
+                cohort[i] = _newAddress;
+
+                address[] memory membersToAdd;
+                address[] memory membersToRemove;
+                membersToAdd[0] = _newAddress;
+                membersToRemove[0] = _currentAddress;
+                _dispatchUpdateMembers(membersToAdd, membersToRemove);
+            }
+        }
+        rotationNonce += 1;
+    }
+
+    /// @notice Get data to sign for a key rotation. Uses an incremented nonce
+    /// @param _currentAddress Address to rotate out
+    /// @param _newAddress Address to rotate in
+    function getRotateDataToSign(address _currentAddress, address _newAddress)
+        public
+        view
+        returns (bytes32)
+    {
+        return keccak256(abi.encodePacked(_currentAddress, _newAddress, rotationNonce));
+    }
+
+    /// @notice Verify signature over data
+    /// @param data Signed data
+    /// @param signature over data
+    /// @param account signer
+    /// @return true if signature is valid and from account
+    function _verify(bytes32 data, bytes memory signature, address account)
+        public
+        pure
+        returns (bool)
+    {
+        return data.toEthSignedMessageHash().recover(signature) == account;
```

Comment:
I think its better to use [EIP-712](https://eips.ethereum.org/EIPS/eip-712) typed hash